docs/features/software-catalog/kubernetes-in-backstage.md

```diff
@@ -0,0 +1,131 @@
+
+### clusterLocatorMethods
+
+This is used to determine where to retrieve cluster configuration from.
+
+Currently, the only valid serviceLocatorMethod is:
+
+#### config
+
+This clusterLocatorMethod will read cluster information in from config
+
+##### clusters
+
+Used by the `config` `clusterLocatorMethods` to construct Kubernetes clients.
+
+###### url
+
+The base url to the Kubernetes control plane. Can be found by using the
+`Kubernetes master` result from running the `kubectl cluster-info` command.
+
+###### name
+
+A name to represent this cluster, this must be unique within the `clusters`
+array. Users will see this value in the Service Catalog Kubernetes plugin.
+
+###### authProvider
+
+This determines how the Kubernetes client authenticate with the Kubernetes
+cluster. Valid values are:
+
+| Value            | Description                                                                                                                                                                                                                       |
+| ---------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `serviceAccount` | This will use a Kubernetes [service account](https://kubernetes.io/docs/reference/access-authn-authz/service-accounts-admin/) to access the Kubernetes API. When this is used the `serviceAccountToken` field should also be set. |
+| `google`         | This will use a user's google auth token from the [google auth plugin](https://backstage.io/docs/auth/) to access the Kubernetes API.                                                                                             |
+
+###### serviceAccount (optional)
+
+The service account token to be used when using the `authProvider`,
+`serviceAccount`.
+
+## RBAC
+
+The current RBAC permissions required are read-only cluster wide, for the
+following objects:
+
+- pods
+- services
+- configmaps
+- deployments
+- replicasets
+- horizontalpodautoscalers
+- ingresses
+
+## Surfacing your Kubernetes components as part of an entity
+
+There are two ways to surface your kubernetes components as part of an entity.
+The label selector takes precedence over the annotation/service id.
+
+### Common `backstage.io/kubernetes-id` label
+
+#### Adding the entity annotation
+
+In order for Backstage to detect that an entity has Kubernetes components, the
+following annotation should be added to the entity.
+
+```yaml
+annotations:
+  'backstage.io/kubernetes-id': dice-roller
+```
+
+#### Labeling Kubernetes components
+
+In order for Kubernetes components to show up in the service catalog as a part
+of an entity, Kubernetes components must be labeled with the following label:
+
+```yaml
+'backstage.io/kubernetes-id': <ENTITY_NAME>
```

Comment:
wait is this right? isn't it supposed to be a made-up identifier such as dice-roller no matter what the entity name is?

Reply:
This is the current logic afaik, it's not the ideal logic.

Reply:
Ok!